client/src/js/components/bhDateInterval.js

```diff
@@ -38,8 +39,10 @@ function bhDateInterval(moment, bhConstants) {
   vm.search = search;
   vm.clear = clear;
 
-
   vm.$onInit = function $onInit() {
+    // specify if clear button can be displayed
+    vm.canClear = !angular.isUndefined(vm.canClear) ? Boolean(vm.canClear) : true;
```

Comment:
Also, I don't think the `Boolean` call does anything.  Try this in your console:
```js
var f = 'false';
console.log(Boolean(f)); // => true
```

Since you use `false` later, that means it is passing it in by value, not as a string to parse.  You can probably just do:
```js
vm.canClear = (vm.canClear || true);
```
to the same effect.